torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1470,6 +1470,22 @@ def sample_inputs_linalg_matrix_power(op_info, device, dtype, requires_grad, **k
         for n in [-4, -2, -1]:
             yield SampleInput(make_arg_fullrank(*size), args=(n,))
 
+def sample_inputs_pow(op, device, dtype, requires_grad, **kwargs):
```

Comment:
Since we're adding a custom sample input function, maybe we want to add a test case that checks the complex + whatever -> complex type promotion case?

Reply:
By adding the np.power ref I see that I have to skip a test_type_promotion skip. Does that overlay/cover the need for this?

Reply:
Why does the adding the ref require skipping test_type_promotion?

Reply:
Sorry I worded that wrong, once I added the ref I found that I need to add a skip for test_type_promotion. 

I did some further digging and here: https://github.com/pytorch/pytorch/blob/f6bbecf8b503eb6562d52b66124a49b45a57f14a/test/test_binary_ufuncs.py#L192
Even though a sampling function is passed into the op the default generator is used. So I was able to repass in the rhs kwarg.  However for the other - small, large, extremal these are hard coded special values. I could not figure out a way of specifying to not raise the ints to negative exponents leading to the the skips.  

The complex skips seem to be comparison of nans again. I am not entirely sure if this is caused by overflow/ underflow but they seem to be consistent with numpy.